### DISCUSSION TITLE
Limit formdata file size

### DISCUSSION THREAD

**Discussion by SupertigerDev** [kartikk221/hyper-express#300] [Q&A] [answered]
How can I limit file upload size?

**Comment by kartikk221** [comment, marked as answer]
Hello, there are multiple ways to approach this:
- Specify a global `max_body_length` when creating the `HyperExpress.Server` instance to restrict global Request body size.
- Specify a route-specific `max_body_length` when creating the route with options to restrict that specific route's body size.
- Specify a custom limit within the `BusboyConfig.limits` options when calling `Request.multipart(config)` to specify a custom limiter within the multipart parser itself.

Overall, I would recommend handling this at the Server/Route level since then HyperExpress will gracefully flush any incoming additional data and handle closing of the request for you for oversized requests.

**Reply by SupertigerDev** [reply]
If I want to restrict size for certain fields, is my only option `Request.multipart(config)`?

**Reply by SupertigerDev** [reply]
![image](https://github.com/user-attachments/assets/ce2d0154-7f16-4693-8345-3e5db3f5f17b)
PS: seems like `BusboyConfig` has no typescript types

**Reply by SupertigerDev** [reply]
@kartikk221 when i use `max_body_length` how do i return a custom error message? it currently returns nothing in the body
